app/src/main/java/org/andstatus/todoagenda/task/AbstractTaskProvider.java

```diff
@@ -28,17 +28,23 @@ protected void initialiseParameters() {
 
     public abstract List<TaskEvent> getTasks();
 
-    protected DateTime getDueDate(Long dueMillis) {
-        if (dueMillis == null) {
-            return now.withTimeAtStartOfDay();
+    protected DateTime getDueDate(Long dueMillis, Long startMillis) {
```

Comment:
"getDueDate" should return due date. Logic of choosing, which date to take for showing in the widget, should be clear and explicit and definitely shouldn't be called "getDueDate".